### PR TITLE
Update support data for StaticRange() constructor

### PR DESCRIPTION
**Pull request by sideshowbarker**
Test results for the StaticRange() constructor are here:

https://wpt.fyi/results/dom/ranges/StaticRange-constructor.html

Those results indicate Firefox and Safari have support, but Chrome and Edge don’t.

### CODE REVIEW

File: api/StaticRange.json [hunk 1]
```diff
@@ -54,19 +54,19 @@
...
-              "version_added": "18"
+              "version_added": null
             },
             "firefox": {
-              "version_added": false
+              "version_added": true
             },
             "firefox_android": {
-              "version_added": false
+              "version_added": true
```
Comment (Elchi3) [inline api/StaticRange.json]:
```suggestion
              "version_added": false
```

**Comment by sideshowbarker** [comment]
> Also, weirdly, we've added supported in 69, but then also "In Firefox, `StaticRange` can currently only be used by browser-internal code or code with enhanced permissions; it is not yet exposed to the web.", so it isn't really supported if this is true and we should rather say version_added: false.

Yeah I saw that statement too and am wondering if it’s accurate. The reason I say that is, if you try the test case at http://web-platform-tests.live/dom/ranges/StaticRange-constructor.html yourself in Firefox, I think you’ll find it’s all green. And the code of that test case isn’t running with any enhanced permissions (and also of course not testing anything that would be browser-internal only).

**Comment by Elchi3** [comment]
For me the tests all fail. Maybe you are in Nightly which often has additional features enabled by default. However, BCD is about what is shipping in release to real users and not to nightly users or users who have additional things enabled. But let's wait what @a2sheppy says on the matter.

**Comment by sideshowbarker** [comment]
> For me the tests all fail. Maybe you are in Nightly which often has additional features enabled by default.

Aha, I am, yes

Note that the results at https://wpt.fyi/results/dom/ranges/StaticRange-constructor.html are also for Nightly

> However, BCD is about what is shipping in release to real users and not to nightly users or users who have additional things enabled.

Ah yeah, that’s an important difference

File: api/StaticRange.json [hunk 2]
```diff
@@ -54,16 +54,16 @@
           "description": "<code>StaticRange()</code> constructor",
           "support": {
             "chrome": {
-              "version_added": "60"
+              "version_added": null
```
Comment (jpmedley) [inline api/StaticRange.json]: Chrome, Chrome Android, and WebView Android should say `false`.

Comment (sideshowbarker) [reply api/StaticRange.json]: > Chrome, Chrome Android, and WebView Android should say `false`.

OK, thanks — made that change

**Comment by jpmedley** [comment]
Approval for Chrome, Chrome Android, and Android WebView data.

File: api/StaticRange.json [hunk 3]
```diff
@@ -54,16 +54,16 @@
...
-              "version_added": "60"
+              "version_added": false
             },
             "chrome_android": {
-              "version_added": "60"
+              "version_added": false
             },
             "edge": {
-              "version_added": "18"
+              "version_added": null
```
[the remark this answers is not in the document]
Comment (sideshowbarker) [reply api/StaticRange.json]: > After some further review, it turns out that Edge **does** actually have support since version 18. 

If you run the test at http://web-platform-tests.live/dom/ranges/StaticRange-constructor.html in Edge, does it pass?

**Comment by sideshowbarker** [comment]
> Since Chrome is false, Opera and Opera Android should also be `false`.

OK, made that change and pushed it to the PR branch

File: api/StaticRange.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sideshowbarker) [reply api/StaticRange.json]: > If you run the test at http://web-platform-tests.live/dom/ranges/StaticRange-constructor.html in Edge, does it pass?

Or http://w3c-test.org/dom/ranges/StaticRange-constructor.html

[the remark this answers is not in the document]
Comment (vinyldarkscratch) [reply api/StaticRange.json]: Can't access that first link, but the second link shows that it's not working.  Sorry, never mind!  Let's set it to `false` then.

File: api/StaticRange.json [hunk 4]
```diff
@@ -54,16 +54,16 @@
...
             },
             "firefox": {
-              "version_added": false
+              "version_added": true
```
Comment (vinyldarkscratch) [inline api/StaticRange.json]: #5104 sets this to "71" as per https://bugzilla.mozilla.org/show_bug.cgi?id=1575980, can we do the same?

File: api/StaticRange.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sideshowbarker) [reply api/StaticRange.json]: > Can't access that first link, but the second link shows that it's not working.

Yeah, they recently changed the domain name (but neglected to set up redirect from the old one):

http://wpt.live/dom/ranges/StaticRange-constructor.html

> Sorry, never mind! Let's set it to `false` then.

OK

File: api/StaticRange.json (hunk 4 above)
Comment (sideshowbarker) [reply api/StaticRange.json]: > "71"

OK, thanks — done

File: api/StaticRange.json [hunk 5]
```diff
@@ -72,22 +72,22 @@
...
-              "version_added": "47"
+              "version_added": false
             },
             "opera_android": {
-              "version_added": "44"
+              "version_added": false
             },
             "safari": {
-              "version_added": false
+              "version_added": true
```
Comment (vinyldarkscratch) [inline api/StaticRange.json]: Looks like Safari actually doesn't support this, based upon running http://w3c-test.org/dom/ranges/StaticRange-constructor.html in the latest release (13.0.3).  Let's set this to `false` as well?

```suggestion
              "version_added": false
```

File: api/StaticRange.json [hunk 6]
```diff
@@ -72,22 +72,22 @@
...
             },
             "safari_ios": {
-              "version_added": false
+              "version_added": true
```
Comment (vinyldarkscratch) [inline api/StaticRange.json]:
```suggestion
              "version_added": false
```

File: api/StaticRange.json (hunk 5 above)
Comment (sideshowbarker) [reply api/StaticRange.json]: > Looks like Safari actually doesn't support this, based upon running http://w3c-test.org/dom/ranges/StaticRange-constructor.html in the latest release (13.0.3). Let's set this to `false` as well?

It’s supported in current Safari Technology Preview. That means it’ll ship in Safari 13.1. So I’ve gone ahead and set the value to `"13.1"`

Comment (vinyldarkscratch) [reply api/StaticRange.json]: "13.1" is not valid for Safari iOS, however.  Can we set Safari iOS to `false`?

Comment (sideshowbarker) [reply api/StaticRange.json]: > "13.1" is not valid for Safari iOS, however. Can we set Safari iOS to `false`?

oops, yeah, thanks for catching that — changed now

Comment (vinyldarkscratch) [reply api/StaticRange.json]: Ah, apparently we don't have Safari Desktop "13.1" in our data either (I could have sworn we had added it)...

Comment (sideshowbarker) [reply api/StaticRange.json]: > Ah, apparently we don't have Safari Desktop "13.1" in our data either (I could have sworn we had added it)...

Oh… Should I create a PR for that?

Comment (vinyldarkscratch) [reply api/StaticRange.json]: After further review, it appears that it was decided not to include Safari 13.1 in general (see #4933).

**Comment by vinyldarkscratch** [comment]
Hey @sideshowbarker, do you plan to return to this PR?

**Comment by sideshowbarker** [comment]
> Hey @sideshowbarker, do you plan to return to this PR?

What else needs to be done for it?

**Comment by vinyldarkscratch** [comment]
We still need to set Safari Desktop to `false`.

**Comment by sideshowbarker** [comment]
> We still need to set Safari Desktop to false.

I'd prefer not to do that, but to instead indicate in some way that the engine does in fact already have support, and that it has shipped in a release that's available to developers

**Comment by vinyldarkscratch** [comment]
We could add a note indicating such support, but it was decided not to include Safari 13.1 until an official release.

**Comment by sideshowbarker** [comment]
> We could add a note indicating such support,

What form would such a note take? The normal way is a `notes` field to accompany the `version_added` value, right? But I don’t usually see that done if the `version_added` value is `false`.

So for me at least, it seems like we still have the issue of needing to set some `version_added` value other than `false` — because `false` isn’t accurate in this case.

> but it was decided not to include Safari 13.1 until an official release.

I understand that I have no strong opinion that the value needs to be "13.1". But the fact remains that there is a shipping version of the browser that has support enabled (albeit a different version than the stable channel). Having the data indicate otherwise isn’t accurate. 

Maybe for cases like this we can consider have an unversioned generic value like `dev` or `prerelease` or something?

**Comment by ddbeck** [comment]
Hi, @sideshowbarker.  Hopefully I can clear some questions up here.

> > We could add a note indicating such support,
> 
> What form would such a note take? The normal way is a `notes` field to accompany the `version_added` value, right? But I don’t usually see that done if the `version_added` value is `false`.

My inclination is to not bother with the note, since it's likely to soon be out of date. It's not the complete picture, but for most web developers, `false` is more or less the information that they need: the feature's not supported in a public release.

I got a bit wordy after this, but the paragraph above is the important bit. 😃 

> I understand that I have no strong opinion that the value needs to be "13.1". But the fact remains that there is a shipping version of the browser that has support enabled (albeit a different version than the stable channel). Having the data indicate otherwise isn’t accurate.

BCD isn't and can't be perfectly up-to-date or comprehensive. Our focus has been on stable versions of supported browsers, as those are the versions that most impact web developers. Beta and pre-release features are certainly interesting but that data is subject to rapid change. Unless the browser vendor has a clear process for moving those features into stable releases—and the vendors actually communicate that process—then we've been reluctant to include pre-release data speculatively.

Compared to some other browsers, the Safari release process is opaque and we have had very little participation on BCD from the vendor. Just because a feature is in a TP now doesn't mean it will be in the next public release of Safari, or won't be backported to an earlier release line (e.g., some point release of 13.0—this is a somewhat common occurrence with Safari).

I suppose that's a rather long-winded way of saying that I don't trust Safari TPs and betas to be indicative of the final version number for a feature. 😆

> Maybe for cases like this we can consider have an unversioned generic value like `dev` or `prerelease` or something?

#338 is open for this. We don't yet have an answer to the question. It's a complicated and hasn't got to the top of our to-do list yet. It may be something we resolve latter, but it won't happen in the course of this PR, unfortunately.

Let me know if there's anything else I can do to help out on this PR. Thank you!

**Comment by sideshowbarker** [comment]
@vinyldarkscratch Thanks much for making the fix and for getting this merged. I had been drafting a response to @ddbeck’s comment but hadn’t been able to make time yet to finish it. Sorry. Anyway, in hindsight was gonna end up being too long. So (for the record here), I’ll try to respond here now, and keep it as brief as I can (though it’s ended up to still be not so brief…).

@ddbeck Thanks much for taking time to respond on this. While I very much agree with having a focus on stable versions of supported browsers, as those are the versions that most impact web developers, I don’t think from that it necessarily follows that the data in BCD needs to be limited only to stable browsers in a case like this.

Part of the context here is that in the W3C/WHATWG world, we are now using the BCD data as part of the work we do in evaluating browser conformance to specs. In particular, we’ve been adding mechanisms to our spec-publishing tools that annotate the specs with links to MDN and with data from BCD. And for our purposes, what we care about is if there’s *any* version of a particular browser which has implemented support for a particular feature — even if it’s only in the current trunk build and not yet in a stable version widely used by developers.

However, I realize that need diverges from the focus that the BCD maintainers have for the data (which, again, is a focus I largely agree with) — and given that for the work I’m doing I’m already using a downstream BCD fork (at https://github.com/w3c/browser-compat-data, with the biggest difference being that the fork adds `spec_url` data to all features for which it’s available), then in cases like this I will just patch BCD on that fork (e.g., in this case, make the change to `"version_added": "13.1"`), and going forward, maintain that patch downstream (until such time as the feature gets into a stable version and so we can actually make the change directly in upstream BCD).

Anyway, thanks again to everybody who’s put time into this issue — especially to @vinyldarkscratch for staying so on top of things, as usual — and sorry again for my delays in getting back to this at the points where the issue resolution was waiting on me for a response.

**Comment by ddbeck** [comment]
@sideshowbarker that's a really interesting use case! Thank you for filling me in on the background. We definitely want to help consumers of BCD and knowing how you're using pre-release data is really important and helpful.

@Elchi3 interesting comment above, if you haven't seen it already.
